test/PushNotificationTests/APITests.cpp

```diff
@@ -0,0 +1,184 @@
+        }
+
+        TEST_METHOD(ChannelRequestUsingRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://ChannelRequestUsingRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(MultipleChannelRequestUsingSameRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://MultipleChannelRequestUsingSameRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(MultipleChannelRequestUsingMultipleRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://MultipleChannelRequestUsingMultipleRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(ThreeChannelRequestUsingSameRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://ThreeChannelRequestUsingSameRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(BackgroundActivation)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+            auto LocalBackgroundTask = winrt::create_instance<winrt::Windows::ApplicationModel::Background::IBackgroundTask>(c_comServerId, CLSCTX_ALL);
+            auto mockBackgroundTaskInstance = winrt::make<MockBackgroundTaskInstance>();
+            LocalBackgroundTask.Run(mockBackgroundTaskInstance);
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(RegisterActivator)
```

Comment:
I think we should combine the REgistActivator and UnregisterActivator test. It makes sense to combine the 2 because Unregister takes the token from Register.